packages/node/src/dev-server.mts

```diff
@@ -46,12 +52,31 @@ async function createEventHandler(
     );
   }
 
+  const content = await readFile(entrypointPath, 'utf8');
+
+  const isStreaming =
+    staticConfig?.supportsResponseStreaming ||
+    (await hasWebHandlers(async () => parseCjs(content).exports)) ||
+    (await hasWebHandlers(async () =>
+      init.then(() => parseEsm(content)[1].map(specifier => specifier.n))
+    ));
```

Comment:
<!-- Questions comments seek clarification.  -->
**question (soft-blocking):** ‏Can we expand this very complex statement?

```suggestion
  function hasCjsWebHandlers(content) {
    return hasWebHandlers(async () => 
      parseCjs(content).exports
    );
  }
  
  function hasEsmWebHandlers(content, init) {
    return hasWebHandlers(async () =>
      init.then(() => {
        // TODO: what is this part of the result?
        // What name can we give it?
        const namedThing = parseEsm(content)[1];
        if (!namedThing) {
          // TODO: What's an appropriate error here?
          throw new Error('Some Error');
        }
        
        // TODO: why do we need to do this?
        // Can we add a comment about what this does?
        return namedThing.map(specifier => specifier.n)
      })
    );
  }

  const isStreaming = staticConfig?.supportsResponseStreaming
    || await hasCjsWebHandlers(content)
    || await hasEsmWebHandlers(content, init);
```

I think the next person to read through this would appreciate some more explicit logic, names, and comment or two.